src/shogun/classifier/mkl/MKLOneClass.cpp

```diff
@@ -21,5 +21,8 @@ float64_t CMKLOneClass::compute_sum_alpha()
 void CMKLOneClass::init_training()
 {
 	ASSERT(svm)
-	ASSERT(svm->get_classifier_type() == CT_LIBSVMONECLASS)
+	ASSERT(svm->get_machine_problem_type() == PT_CLASS)
+	// TODO: FIXME this currently supports LibSVMOneClass only
+	// should also support SVMLightOneClass
+	ASSERT(std::string(svm->get_name()) != "SVMLightOneClass")
```

Comment:
i would refrain myself for this type checking... you can use `std::type_index(typeid)`

Reply:
will that work if the type `SVMLightOneClass` is not known? (plugins)

Reply:
this class already obviously depends on `SVMLightOneClass` so yes then it should work...

Reply:
The dependency is kinda unnecessary through the ctor only if you look at it (and should be removed imo). In general, the MKL stuff should work with pretty much any SVM solver as long as there is some interface defined.

Reply:
well mkl doesn't work with anything else but svmlight* stuff ... just saying.